pkg/proxy/proxy.go

```diff
@@ -200,11 +206,16 @@ func doTokenRequest(ctx context.Context, clientID, resource, tenantID, authority
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to acquire token")
 	}
+	expiresIn, err := parseExpiresIn(strconv.FormatInt(result.ExpiresOn.UTC().Unix()-time.Now().UTC().Unix(), 10))
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to parse expires in")
+	}
 
 	return &token{
 		AccessToken: result.AccessToken,
 		Resource:    resource,
 		Type:        "Bearer",
+		ExpiresIn:   expiresIn,
```

Comment:
Reading the code I see that clock skew is not an issue here because `ExpiresOn` is computed locally.  I think this approach is simpler (I took off 10 seconds to account for the current time changing between calls):

```suggestion
		ExpiresIn:   json.Number(strconv.FormatInt(int64(time.Until(result.ExpiresOn)/time.Second) - 10, 10)),
```

Reply:
Updated and removed `parseExpiresIn` as we don't need to handle different format. As per the AAD doc, `expires_in` is in seconds.